### ISSUE TITLE
overflow of .twelve.columns div in fixed width tablet portrait view

### ISSUE THREAD

**Issue by dirkolbrich** [atomicpages/skeleton-sass#28]
A `.twelve .column.`wide `<div>` is too wide in iPad portrait screen. It overflows the surrounding `.container`.

Following html setup. Theme is `Fresh`with `$is-fluid: false;`setup:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>sample title</title>
    <link rel="stylesheet" href="/css/main.css">
</head>

<body>
    <!-- .container is main centered wrapper -->
    <div class="container">
        <div class="row">
            <div class="twelve columns">
                <p>This column is too wide for surrounding container in browser width 960 - 1100px</p>
            </div>
        </div>
    </div>
</body>

</html>
```

## Problem

Right in between the sweet spot of 960 - 1100px browser width (iPad is 1024px on portrait), the `.twelve.colums`div is wider than the surrounding `.container` div. I think the reason is `.twelve.colums`has no `max-width`set at that point.

This is caused by the `@media only screen and (min-width: $tablet-width) and (max-width: $base-width - 1) {}` query in `/themes/fresh/_grid.scss:34` as it stops at 959px browser width.

```css
/* TABLET (PORTRAIT) */
@media only screen and (min-width: 768px) and (max-width: 959px) {
  .container {
    width: 768px;
  }
  /* ... */
  .container .twelve.columns {
    width: 748px;
  }
}
```

From there on, the other smaller media queries kick in:

```css
/* For devices larger than 400px */
@media (min-width: 400px) {
  .container {
    width: 85%; /* results to 816px width at 960px browser, needs browser at 1100px width until it is wide enough for .twelve.columns */
  }
}

/* For devices larger than 550px */
@media (min-width: 550px) {
  .container { /* no width set, is by 85% */ }
  .container .twelve.columns {
    width: 940px;
  }
}
```

## Proposed solution

Give all columns a `max-width: 100%` in `/core/_dependencies.scss:96`:

```scss
.#{numToString($i)}.columns { 
  width: $colWidth + (($colWidth + $gutterWidth) * ($i - 1));
  max-width: 100%;
}
```

Yet I don`t know for sure, if this would cause any side effects on other settings.

**Comment by dirkolbrich** [comment]
better solution to set `max-width`for all `.columns` at `/core/_dependencies.scss:83`:

```scss
@mixin _fixedGrid($width, $colWidth, $gutterWidth, $colCount) {
	.container {
		.column,
		.columns {
			max-width: 100%;
			margin: {
				left: $gutterWidth / 2;
				right: $gutterWidth / 2;
			};
		}
	}
	/* The Grid */
       /* ... */
}
```

**Comment by atomicpages** [comment]
@dirkolbrich can you help me out with an exact repro? Here are some good sites for sharing code:

* http://codepen.io/
* http://jsbin.com/
* https://jsfiddle.net/

**Comment by dirkolbrich** [comment]
Ok, here is the CodePen [testing skeleton-sass fixed grid](https://codepen.io/dirkolbrich/pen/RVVggw). 

Setup is right from the skeleton-scss README, no additional styles, except the gray-boxes.
max-width 960px, 12 columns

There are several issues with the base fixed-grid:
1. at screen width 320px
    - at small screen width the `.one.colum` is wider than the `.one.colums` div -> #29
    - `.column` and `.columns` divs overflow the surrounding `.container` -> #31
    - `.column` and `.columns` divs are not centered -> compare to `.full-width` column
2. at screen width 400px
    - `.container` div has no padding, below width 400px a `padding: 20px;`, this gives a little jump at changing screen size
3. screen width between 960px und 1129px
    - `.columns` divs `.one` to `.eleven` will be broken over, if added to a sum of twelve columns
    - `.twelve` and `.full-width` overflow `.container`ans screen -> #28

**Comment by dirkolbrich** [comment]
added basic layouts and offset to the CodePen.

Following issue in addition to the above:
4. `.one-half` columns have no css class assigned

**Comment by atomicpages** [comment]
@dirkolbrich thanks for all your hard work identifying the issues! I alongside our small Skeleton Sass community greatly appreciates it! I believe all issues you identified are resolved in release 3.1.4

**Comment by dirkolbrich** [comment]
Thanks. I am just finger pointing. You did the work. 
Thank you for this repo.
